src/main/java/gift/service/WishListService.java

```diff
@@ -0,0 +1,65 @@
+package gift.service;
+
+import gift.exception.RepositoryException;
+import gift.model.WishList;
+import gift.model.WishListDTO;
+import java.util.List;
+import java.util.stream.Collectors;
+import org.springframework.stereotype.Service;
+
+@Service
+public class WishListService {
+
+    private final WishListRepository wishListRepository;
+
+    public WishListService(WishListRepository wishListRepository) {
+        this.wishListRepository = wishListRepository;
+    }
+
+    public void createWishList(WishListDTO wishListDTO) {
+        WishList wishList = new WishList(wishListDTO.email(), wishListDTO.memberId(),
+            wishListDTO.productName(), wishListDTO.quantity());
+        wishListRepository.save(wishList);
+    }
+
+    public List<WishListDTO> getAllWishList() {
+        List<WishList> wishlists = wishListRepository.findAll();
+        return wishlists.stream()
+            .map(this::convertToDTO)
+            .collect(Collectors.toList());
+    }
+
+    public List<WishListDTO> getWishListById(long memberId) {
+        List<WishList> wishlists = wishListRepository.findByMemberId(memberId)
+            .orElseThrow(() -> new RepositoryException("해당 사용자의 위시 리스트는 비어 있습니다."));
+        return wishlists.stream()
+            .map(this::convertToDTO)
+            .collect(Collectors.toList());
+    }
+
+    public void updateWishListQuantity(WishListDTO wishListDTO) {
+
+        WishList currentWishList = wishListRepository.findByMemberIdAndProductName(
+            wishListDTO.memberId(),
+            wishListDTO.productName()).orElseThrow(() -> new RepositoryException(
+            wishListDTO.email() + "의 사용자의 위시 리스트에서 " + wishListDTO.productName()
+                + "을(를) 찾지 못했습니다."));
```

Comment:
예외메시지 분리와 코드 Line을 정리하면 이런 형태로도 코드를 바꿀 수 있을 것 같아요.

```suggestion
private final String NOT_FOUND_WISHLIST = "%s 의 사용자의 위시 리스트에서 %s 을(를) 찾지 못했습니다.";
    public void updateWishListQuantity(WishListDTO wishListDTO) {
        WishList currentWishList = wishListRepository.findByMemberIdAndProductName(
                        wishListDTO.memberId(),
                        wishListDTO.productName()
                )
                .orElseThrow(() -> new RepositoryException(
                        NOT_FOUND_WISHLIST.formatted(wishListDTO.email(), wishListDTO.productName()))
                );
```